fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rewrite/ExpressionOfPlanRewrite.java

```diff
@@ -0,0 +1,136 @@
+import org.apache.doris.nereids.trees.plans.logical.LogicalProject;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * expression of plan rewrite rule.
+ */
+public class ExpressionOfPlanRewrite implements PlanRuleFactory {
+    ExpressionRuleExecutor rewriter;
+
+    public ExpressionOfPlanRewrite(ExpressionRuleExecutor rewriter) {
+        this.rewriter = Objects.requireNonNull(rewriter, "rewriter is null");
+    }
+
+    @Override
+    public List<Rule<Plan>> buildRules() {
+        return ImmutableList.of(new ProjectExpressionRewrite().build(), new AggExpressionRewrite().build(),
+                new FilterExpressionRewrite().build(), new JoinExpressionRewrite().build());
+    }
+
+    @Override
+    public RulePromise defaultPromise() {
+        return RulePromise.REWRITE;
+    }
+
+    private class ProjectExpressionRewrite extends OneRewriteRuleFactory {
+        @Override
+        public Rule<Plan> build() {
+            return logicalProject().then(project -> {
+                List<NamedExpression> origin = project.getProjects();
+                for (int i = 0; i < origin.size(); i++) {
+                    if (origin.get(i) instanceof Alias) {
+                        Alias alias = (Alias) origin.get(i);
+                        Expression newExpr = rewriter.rewrite((alias).child());
+                        origin.set(i, (NamedExpression) alias.withChildren(Lists.newArrayList(newExpr)));
+                    }
+                }
+                return new LogicalProject<>(origin, project.child());
+            }).toRule(RuleType.REWRITE_PROJECT_EXPRESSION);
+        }
+    }
+
+    private class FilterExpressionRewrite extends OneRewriteRuleFactory {
+        @Override
+        public Rule<Plan> build() {
+            return logicalFilter().then(filter -> {
+                Expression newExpr = rewriter.rewrite(filter.getPredicates());
+                if (newExpr.equals(filter.getPredicates())) {
+                    return filter;
+                }
+                return new LogicalFilter<>(newExpr, filter.child());
+            }).toRule(RuleType.REWRITE_FILTER_EXPRESSION);
+        }
+    }
+
+    private class AggExpressionRewrite extends OneRewriteRuleFactory {
+        @Override
+        public Rule<Plan> build() {
+            return logicalAggregate().then(agg -> {
+                List<Expression> groupByExprs = agg.getGroupByExpressionList();
+                List<Expression> newGroupByExprs = rewriter.rewrite(groupByExprs);
+
+                List<NamedExpression> outputExpressions = agg.getOutputExpressionList();
+                List<NamedExpression> newOutputExpressions = Lists.newArrayList();
+                for (NamedExpression expr : outputExpressions) {
+                    if (expr instanceof Alias) {
+                        Alias alias = (Alias) expr;
```

Comment:
we don't need to do cast explicitly.
BTW, why we could not just rewrite all expressions?